src/reanimated2/NativeMethods.ts

```diff
@@ -21,36 +19,42 @@ export interface MeasuredDimensions {
   pageY: number;
 }
 
-const isNativeIndefined = shouldBeUseWeb();
+const isNativeUndefined = shouldBeUseWeb();
 
 export function measure(
   animatedRef: RefObjectFunction<Component>
-): MeasuredDimensions {
+): MeasuredDimensions | null {
   'worklet';
-  if (!_WORKLET || isNativeIndefined) {
+  if (!_WORKLET || isNativeUndefined) {
     console.warn(
-      '[reanimated.measure] method cannot be used for web or Chrome Debugger'
+      '[Reanimated] measure() cannot be used for web or Chrome Debugger'
     );
-    return {
-      x: NaN,
-      y: NaN,
-      width: NaN,
-      height: NaN,
-      pageX: NaN,
-      pageY: NaN,
-    };
+    return null;
   }
   const viewTag = animatedRef();
-  const result = _measure(viewTag);
-  if (result.x === -1234567) {
-    throw new Error(`The view with tag ${viewTag} could not be measured`);
-  }
-  if (isNaN(result.x)) {
+  if (viewTag === -1) {
     console.warn(
-      'Trying to measure a component which gets view-flattened on Android. To disable view-flattening, set `collapsable={false}` on this component.'
+      `[Reanimated] The view with tag ${viewTag} is not a valid argument for measure()`
     );
+    return null;
+  }
+  try {
+    const result = _measure(viewTag);
+    if (result.x === -1234567) {
+      throw new Error(
+        `The view with tag ${viewTag} returned an invalid measurement response`
+      );
```

Comment:
Do we really want to throw an error here? In all other places it's `console.warn` and `return null;`.

Reply:
I also suggest using `console.warn` everywhere.

Reply:
This doesn't actually throw outside of the main `measure()` function - it skips execution down to the `catch` block, which uses the error in a `console.warn`. 

But I see how this is less readable, so I can just make it an inline warning.

Reply:
Thanks!